app/src/controllers/authenticationController.js

```diff
@@ -75,11 +58,20 @@ export default class AuthenticationController {
    *
    * @param req
    * @param res
+   * @param samlCertFile
+   * @param spidStrategy
    */
-  metadata(req: express$Request, res: express$Response) {
-    const certPath = process.env.SAML_CERT_PATH || "./certs/cert.pem";
-    winston.log("info", "Reading SAML certificate file from %s", certPath);
+  metadata(
+    req: express$Request,
+    res: express$Response,
+    samlCertFile: string,
```

Comment:
can't `samlCertFile` and `spidStrategy` be set in the constructor?

either the life of the `AuthenticationController` class spans multiple calls to `metadata` (and I assume `samlCertFile` and  `spidStrategy`) or `AuthenticationController` and the `metadata` method have the same lifecycle (that you don't need a class bust just a single method)

Reply:
let me know you you want to have a quick call to clarify my point further

Reply:
No it's clear, I'll refactor the code a bit to inject all the dependencies directly into the class constructor